

\chapter{Introduction}
\label{chap:introduction}

\section{Motivation}
\label{sec:motivation}
Bioinformatics may be defined as the study and analysis of biological material using computational methods. At least, two important areas of research exist: genomics and proteomics. Genomics comprises the study of genomes. The former aims to look at the discovery and understanding of the complete DNA sequence of organisms with the intent to gain a detailed genetic map. On the other hand, the interest for proteomics completes the knowledge about the next step in which the genetic information, once translated into amino acid sequence, determines the building blocks of life.\\
Proteins, which are the central topic of this work, are fundamental parts of organisms and participate in every process within cells. Many of them have enzymatic functions that catalyze biochemical reactions and are vital to the metabolism. Other proteins have comunicative functions like cell signaling, immune responses and cell cycle. Enzymes are proteins that catalyse biochemical reactions. The function of an enzyme relies on the structure of its active site, a cavity in the protein characterized by a specific shape and size that enable it to bind a certain substrate. A substrate may be another protein, a nucleic acid sequence or any other biochemical compound. Moreover, proteins transport and store other molecules such as oxygen, providing for mechanical support and immune protection. They allow the transmission of nerve impulses and are responsible for the growth control and cell differentiation. \\
Protein functions are determined by the three-\-di\-men\-sio\-nal structure of the protein and the physico-chemical properties of the amino acids at the active site. Therefore, knowledge about the structure of a protein is very important in order to discover its function, understand diseases and design new drugs. The number of known protein sequences is about two orders of magnitude higher than the number of experimentally solved protein structures and only a small subset of these has annotated function. New computational methods to predict the protein structure from its sequence are needed, because experimental methods for the determination of protein structures are time expensive, and fail for some types of proteins, such as membrane proteins. Given a target sequence for which the three-\-di\-men\-sio\-nal structure is desired, actual methods for protein structure prediction generate a huge amount of candidate models. In order to be able to discriminate good from bad protein models, a new class of computational methods has been developed to assess the quality of predicted models.

\section{Objectives}
\label{sec:objectives}
From the comparison of the most effective model quality assessment programs, it is known that clustering-based methods outperform those based on single-models. Methods belonging to the first category evaluate protein models by considering the ensemble, whereas the latter judge every model individually. These investigations suggest that future model quality assessment methods should consider a set of target models at once, while trying to minimize clustering drawbacks such as the loss of the best models which are far away from the core. \\
This work aims at the design and implementation of a new hybrid method for model quality assessment, based on clustering, that weights model scores by using a single-model method. The latter implements a neural network trained on features from the QMEAN statistical potentials, which is another single-model method. In more detail, the first goal is to demonstrate that a machine learning-based approach to the problem can improve the actual QMEAN performance. Different variants including both a single neural network and a neural system, are presented and discussed. The second objective comprises the exploration of new features besides those from QMEAN, again with the intention of outperforming the previous results. These new features consider hydrogen bonds, Gauss integrals and TAP score. Finally, the development of a method based on semi-clustering is treated. In order to improve the accuracy of the assessment, the latter takes into account the predicted model scores given by the previously cited single-model method, the model fraction that has been predicted and finally a categorization of the target sequence based on the predicted models.\\
CASP, which stands for Critical Assessment of Techniques for Protein Structure Prediction, is a community-wide experiment for protein structure prediction. It aims to establish the current state of the art, identifying what progress has been made, and highlighting where future effort may be most productively focused. The clustering-based and individual methods developed in this work are tested on data from the CASP-4 and CASP-8 experiments, showing a significative improvement. The new clustering-based algorithm outperforms all other methods participating in CASP-8, held in December 2008, achieving a global Pearson correlation between predicted quality score and GDT\_TS of $0.9485$ over $117$ targets for a total of $29,064$ models. In detail, the improvement with respect to ModFOLDclust, which is the best participating method and currently the best MQAP in the world, is of $\approx 1\%$ of global correlation. Also, it outperforms the QMEAN clustering-based version by $\approx 1.8\%$. The individual methods based only on the QMEAN features significatively outperform the standard version of QMEAN between $\approx 3\%$ and $\approx 6\%$ of Pearsonâ€™s correlation. Results on the CASP-4 test set are consistent with those in CASP-8.


\section{Outline}
\label{sec:outline}
This thesis is presented in four chapters. The first chapter contains a brief introduction to the performed work. The second chapter provides a more detailed discussion about protein structure and protein structure prediction with the aim to introduce the reader to the problem of model quality assessment. Beginning with the description of proteins, an overview of experimental and computational methods for the determination of three-\-di\-men\-sio\-nal protein structure is presented. A wide presentation on the state of the art for the model quality assessment programs is treated.\\
The third chapter describes the choices, design and implementation of the proposed methods. Starting with the description of QMEAN drawbacks, it discusses the single-model and the clustering-based methods, concluding with the new introduced features.\\
The fourth chapter presents the results achieved by the implemented methods on two test sets and provides a benchmark against the best currently available model quality assessment programs. An exhaustive series of plots is also reported with the motivation to illustrate details where methods fail or achieve good performances. Finally, a conclusive chapter is provided in order to summarize the developed work and define new possible research directions.\\
Two appendices comprising the ideas underlying force fields, statistical potentials and the mathematical theory of Gauss integrals are given as well.

\cleardoublepage



